### PR TITLE
[RFR] Redux without profanity - step 1 (Follow #106)

### PR DESCRIPTION
**Pull request by djhi**
- [x] Cleaned up App container by extracting Navbar into its own component
- [x] Updated some packages including redux-form
- [x] Simplified SignIn and SignUp forms
- [x] Simplified file names by removing prefixes (`userActions` => `actions`, etc.)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -7,6 +7,7 @@
     },
     "extends": "airbnb",
     "rules": {
+        "max-len": "off",
```
Comment (Kmaschta) [inline .eslintrc]: I disagree with it, I open JS files with `vim` relatively often, and it's a pain in the neck to read enormous lines.
Idem for `cat`, `grep` and all very useful console tools, even if we have large screens.

I suggest to increase the limit to 100 or 120 max. It's a good way to keep short variables or functions names too.

**Comment by Kmaschta** [comment]
Great !
I love all the files renaming like `domain/domainReducer` to `domain/reducer`.

File: .eslintrc (hunk 1 above)
Comment (djhi) [inline .eslintrc]: 120 ? Deal :)

Comment (Kmaschta) [inline .eslintrc]: On our project, we have this config:
`"max-len": [2, {"code": 120, "tabWidth": 4, "ignoreUrls": true}],`

**Comment by fzaninotto** [comment]
Can you rebase?

File: src/frontend/js/app/AppNavbar/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+NavbarItem.propTypes = {
+    children: PropTypes.node.isRequired,
+    to: PropTypes.string.isRequired,
+};
+
+const AppNavbar = ({ user, signOut }) => (
+    <nav className="navbar navbar-fixed-top navbar-dark bg-primary">
+        <Link className="navbar-brand" to="/products">New App</Link>
+        <ul className="nav navbar-nav">
+            <NavbarItem to="/products">Home</NavbarItem>
```
Comment (fzaninotto) [inline src/frontend/js/app/AppNavbar/index.js]: let's change to `/`

File: src/frontend/js/app/App.js [hunk 3]
```diff
@@ -94,16 +60,12 @@ App.propTypes = {
...
+const mapStateToProps = state => ({
+    user: state.user,
+});
 
-function mapDispatchToProps(dispatch) {
-    return bindActionCreators({
-        signOut: signOutActions.request,
-    }, dispatch);
-}
+const mapDispatchToProps = dispatch => bindActionCreators({
```
Comment (fzaninotto) [inline src/frontend/js/app/App.js]: Doesn't redux do it automatically if we pass an object as second parameter to `connect()`?

File: src/frontend/js/app/AppNavbar/UserNavbarItem.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                aria-expanded="false"
+                aria-haspopup="true"
+                className="nav-link dropdown-toggle"
+                data-toggle="dropdown"
+                role="button"
+            >
+                {user.email}
+            </a>
+            <div className="dropdown-menu">
+                <a
```
Comment (fzaninotto) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: we could write this one in a single line for brevity

File: src/frontend/js/app/AppNavbar/UserNavbarItem.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { PropTypes } from 'react';
+
+const UserNavbarItem = ({ user, signOut }) => (
+    <ul className="nav navbar-nav pull-xs-right">
+        <li className="nav-item dropdown">
+            <a
+                aria-expanded="false"
```
Comment (fzaninotto) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: are you sure it's not `ariaExpanded`? This is JSX

File: src/frontend/js/app/AppNavbar/index.js (hunk 2 above)
Comment (djhi) [inline src/frontend/js/app/AppNavbar/index.js]: On otherwise react-router will add it the activé class whatever the current route is. We should an home path

File: src/frontend/js/app/routes.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import App from './App';
-import productRoutes from '../product/productRoutes';
-import orderRoutesFactory from '../order/orderRoutes';
-import userRoutes from '../user/userRoutes';
+import productRoutes from '../product/routes';
```
Comment (fzaninotto) [inline src/frontend/js/app/routes.js]: why don't we use route components again?

File: src/frontend/js/product/selectors.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export const getProductById = (state, productId) => (state.product.list.length ? undefined : state.product.list.find(p => p.id === productId));
```
Comment (fzaninotto) [inline src/frontend/js/product/selectors.js]: This should be in a reducer file IMO

Comment (fzaninotto) [inline src/frontend/js/product/selectors.js]: Or maybe I'm misunderstanding what it does

File: src/frontend/js/app/App.js (hunk 3 above)
Comment (djhi) [inline src/frontend/js/app/App.js]: I remember something about this yes. However, I had to add it on another component (can't remember which one) otherwise, the dispatch was not injected.

File: src/frontend/js/app/AppNavbar/UserNavbarItem.js (hunk 5 above)
Comment (djhi) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: I'll check

File: .eslintrc (hunk 1 above)
Comment (djhi) [inline .eslintrc]: didn't know about all those options ! Great :)

File: src/frontend/js/app/AppNavbar/UserNavbarItem.js (hunk 5 above)
Comment (djhi) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: Nope: https://facebook.github.io/react/docs/dom-differences.html

File: src/frontend/js/app/routes.js (hunk 6 above)
Comment (djhi) [inline src/frontend/js/app/routes.js]: Good question, can't remember. If we decide to move to components, I suggest we do so in a separate PR

**Comment by djhi** [comment]
The commits are misleading, the devtool have been removed previously. I already rebased so this ok to merge
